server/routes/manage-csips/controller.ts

```diff
@@ -0,0 +1,107 @@
+import { Request, Response } from 'express'
+import CsipApiService from '../../services/csipApi/csipApiService'
+import PrisonApiService from '../../services/prisonApi/prisonApiService'
+import { setPaginationLocals } from '../../views/partials/simplePagination/utils'
+import { getNonUndefinedProp } from '../../utils/utils'
+import { CsipRecord } from '../../@types/csip/csipApiTypes'
+
+const PAGE_SIZE = 25
+
+export class SearchCsipController {
+  constructor(
+    private readonly csipApiService: CsipApiService,
+    private readonly prisonApiService: PrisonApiService,
+  ) {}
+
+  GET = async (req: Request, res: Response) => {
+    const { page, clear, sort, query, status } = req.query
+
+    if (clear) {
+      req.session.searchCsipParams = {}
+      return res.redirect('manage-csips')
+    }
+
+    req.session.searchCsipParams ??= {}
+
+    if (page || sort || query || status) {
+      if (page) {
+        req.session.searchCsipParams.page = Number.isNaN(Number(page)) ? 1 : Number(page)
+      } else {
+        req.session.searchCsipParams.page = 1
+      }
+
+      if (query || status) {
+        if (query) {
+          req.session.searchCsipParams.query = (query as string).trim() || null
+        } else {
+          delete req.session.searchCsipParams.query
+        }
+        if (
+          status &&
+          [
+            'CSIP_CLOSED',
+            'CSIP_OPEN',
+            'AWAITING_DECISION',
+            'ACCT_SUPPORT',
+            'PLAN_PENDING',
+            'INVESTIGATION_PENDING',
+            'NO_FURTHER_ACTION',
+            'SUPPORT_OUTSIDE_CSIP',
+            'REFERRAL_SUBMITTED',
+            'REFERRAL_PENDING',
+            'UNKNOWN',
+          ].includes(status as string)
+        ) {
+          req.session.searchCsipParams.status = status as CsipRecord['status']
+        } else {
+          delete req.session.searchCsipParams.status
+        }
+        delete req.session.searchCsipParams.sort
+      }
+
+      if (sort) {
+        const [sortingKey, sortingDirection] = (sort as string).split(',')
+        if (
+          ['name', 'location', 'referralDate', 'caseManager', 'nextReviewDate', 'status'].includes(sortingKey ?? '') &&
+          ['asc', 'desc'].includes(sortingDirection ?? '')
+        ) {
+          req.session.searchCsipParams.sort = sort as string
+        }
+      }
+
+      return res.redirect('manage-csips')
```

Comment:
Why do we need to redirect if we have any of the search/query/sort? Can we not just continue with code execution using the values you've just put into searchCsipParams?

Reply:
i want to eliminate potential confusion whether the filter and sort are governed by session or by url query string, so i consume the query strings and populate them into session. 

switching to use url query is technically feasible, but that would mean we need to convert post form data of search string and status filter into query string, and concatenating the query strings for all those page buttons and sort buttons will be a bit horrible.